islandora_video_streaming_panopto.module

```diff
@@ -40,20 +57,256 @@ function islandora_video_streaming_panopto_islandora_sp_videoCModel_islandora_de
+    $content = 'grant_type=password&username='.urlencode(strtolower($username)).'&password='.urlencode($password).'&scope=api';
+    $response = islandora_video_streaming_panopto_request_curl($oauthendpoint, $headers, $content, 'POST', array(200 => 'access_token'));
+  }
+  if ($response) {
+    $token = $response['access_token'];
+  } else {
+    $messages = array(
+      'message' => t('Failed to retrieve $token. $response is @response . $header is @header . $content is @content'),
+      'message_sub' => array(
+        '@response' => $response,
+        '@header' => $header,
+        '@content' => $content,
+      ),
+      'type' => 'watchdog',
+      'severity' => WATCHDOG_ERROR,
+    );
+    $failureMessage['messages'][] = $messages;
+  }
+  return $token;
+}
+
+/**
+ * Function to send a CURL request
+ * @param string $endpoint The URL which is the target of the request
+ * @param array $headers A list of HTTP headers to send
+ * @param string $content The optional body of the request, if applicable
+ * @param string $type The HTTP method to use (defaults to GET)
+ * @param array $expect An optional keyed array of HTTP response codes mapped to an (optional) expected JSON identifier
+ *                      If provided, this is required for the function to return data
+ * @return mixed A string or array.
+ *               If no $expect parameter is given, returns the response body.
+ *               If an $expect key matches with no array value, returns the response body only on a match
+ *               If an $expect key matches and has a value, returns the decoded JSON as an array if that matched value is present as a key
+ */
+function islandora_video_streaming_panopto_request_curl($endpoint, $headers, $content = '', $type = 'GET', $expect = array()) {
+  $cURL = curl_init();
+  curl_setopt($cURL, CURLOPT_URL, $endpoint);
+  curl_setopt($cURL, CURLOPT_RETURNTRANSFER, true);
+  curl_setopt($cURL, CURLOPT_HEADER, 1);
+  curl_setopt($cURL, CURLINFO_HEADER_OUT, true);
+  switch (strtoupper($type)) {
+    case 'POST':
+      curl_setopt($cURL, CURLOPT_POST, true);
+    break;
+    case 'PUT':
+      curl_setopt($cURL, CURLOPT_CUSTOMREQUEST, 'PUT');
+    break;
+    case 'DELETE':
+      curl_setopt($cURL, CURLOPT_CUSTOMREQUEST, 'DELETE');
+    break;
+  }
+  if ($content) {
+    curl_setopt($cURL, CURLOPT_POSTFIELDS, $content);
+  }
+  curl_setopt($cURL, CURLOPT_HTTPHEADER, $headers);
+  $response = curl_exec($cURL);
+  $header_size = curl_getinfo($cURL, CURLINFO_HEADER_SIZE);
+  $response_code = curl_getinfo($cURL, CURLINFO_RESPONSE_CODE);
+  $headerSent = curl_getinfo($cURL, CURLINFO_HEADER_OUT );
+  curl_close($cURL);
+  if ($response === false) {
+    $messages = array(
+      'message' => t("Unable to receive response from curl_exec(): @error "),
+      'message_sub' => array('@error' => curl_error($cURL)),
+      'type' => 'watchdog',
+      'severity' => WATCHDOG_ERROR,
+    );
+    watchdog($messages);
```

Comment:
`watchdog` appears unnecessary here.  Reporting this `$messages` should be handled by the `watchdog` call at the end of the function.

Reply:
Done.